### PR TITLE
Converts - Report components

### PR DESCRIPTION
**Pull request by BronzBierd**
# Fixes or Implements [AB#15782](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/15782)

## Description

1. Convert ReportsView
2. Convert ReportsComponent
    * Immunization
    * Notes
    * Medication
    * Special Authority Requests
    * COVID 19
    * Laboratory Results
    * MSP Visits
    * Hospital Visits

## Testing

- [ ] Unit Tests Updated
- [ ] Functional Tests Updated
- [X] Not Required

## UI Changes
Only a few screens have been used to show the common funcationality:
image
image

Mediations:
image

Filter Applied:
image
image

No records:
image

Skeleton loader:
image

## Items to Review:

-   [General PR Guidelines](https://github.com/bcgov/healthgateway/wiki/PRguidance)

### CODE REVIEW

**Review by mk08sh — COMMENTED** [review]
✅

File: Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+﻿import { DateWrapper } from "@/models/dateWrapper";
+
+export default class DateWrapperSortUtility {
+    /**
+     * Sort comparison test to determine dates in descending order.
+     * @param aDate instance of DateWrapper
+     *  @param bDate instance of DateWrapper
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts:7]:
```suggestion
     * @param bDate instance of DateWrapper
```

File: Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    /**
+     * Sort comparison test to determine dates in ascending order. Using the DateWrapper class to handle the date comparison.
+     * @param aDate valid date string
+     * @param bDate valid date string
+     */
+    static ascendingByString(aDate: string, bDate: string): number {
+        const firstDate = new DateWrapper(aDate);
+        const secondDate = new DateWrapper(bDate);
+
+        return -1 * DateWrapperSortUtility.descending(firstDate, secondDate);
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts:47]:
```suggestion
        return -1 * DateWrapperSortUtility.descendingByString(aDate, bDate);
```

Simplifies code and makes it easier to add additional features to string sorting in the future (e.g. using the `hasTime` option when constructing the DateWrappers).

File: Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/utility/dateWrapperSortUtility.ts:1]: Suggest renaming to DateSortUtility.

File: Apps/WebClient/src/NewClientApp/src/stores/immunization.ts [hunk 3]
```diff
@@ -25,18 +25,11 @@ const defaultImmunizationDatasetState: ImmunizationDatasetState = {
...
-    if (firstDate.isAfter(secondDate)) {
-        return 1;
-    }
-    if (firstDate.isBefore(secondDate)) {
-        return -1;
-    }
-    return 0;
-}
+const immunizationSort = (a: ImmunizationEvent, b: ImmunizationEvent): number =>
+    DateWrapperSortUtility.descendingByString(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/immunization.ts:29]:
```suggestion
    DateWrapperSortUtility.ascendingByString(
```
Note that `firstDate.isBefore(secondDate)` returns -1 in this function, even though the order of the if statements is reversed.

File: Apps/WebClient/src/NewClientApp/src/stores/comment.ts [hunk 4]
```diff
@@ -11,19 +11,13 @@ import { LoadStatus } from "@/models/storeOperations";
...
-    if (firstDate > secondDate) {
-        return 1;
-    }
-    if (firstDate < secondDate) {
-        return -1;
-    }
-    return 0;
-}
+const commentsSort = (a: UserComment, b: UserComment): number =>
+    DateWrapperSortUtility.descendingByString(
```
Comment (MikeLyttle) [inline Apps/WebClient/src/NewClientApp/src/stores/comment.ts:17]:
```suggestion
    DateWrapperSortUtility.ascendingByString(
```

**Review by MikeLyttle — CHANGES_REQUESTED** [review]
I'd also suggest renaming the report component file to match the entry types, similar to the suggestions in AB#15824.

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**